### PR TITLE
Collect JsPlugin instances before starting Jetty

### PR DESCRIPTION
**Pull request by niloc132**
This reverses a DI dependency, preventing requiring that the server start up before plugins can be collected.

Partial #4040

### CODE REVIEW

File: server/jetty/src/main/java/io/deephaven/server/jetty/JettyBackedGrpcServer.java [hunk 1]
```diff
@@ -85,20 +83,15 @@ public class JettyBackedGrpcServer implements GrpcServer {
...
     private final Server jetty;
-    private final JsPlugins jsPlugins;
     private final boolean websocketsEnabled;
 
     @Inject
     public JettyBackedGrpcServer(
             final JettyConfig config,
-            final GrpcFilter filter) {
+            final GrpcFilter filter,
+            final JsPlugins jsPlugins) {
```
Comment (devinrsmith) [inline server/jetty/src/main/java/io/deephaven/server/jetty/JettyBackedGrpcServer.java:92]: This adds a package private class to a public constructor, essentially rendering it non-public to anybody but us. Given the way we construct JettyBackedGrpcServer, it's probably reasonable to make this constructor package private. The other option is to make `JsPlugins` public.

I wish there were a way to force java compilation to fail in these cases?

Comment (devinrsmith) [reply server/jetty/src/main/java/io/deephaven/server/jetty/JettyBackedGrpcServer.java:92]: image

File: server/jetty/src/main/java/io/deephaven/server/jetty/JettyServerModule.java [hunk 2]
```diff
@@ -67,9 +67,15 @@ static ServletAdapter provideGrpcServletAdapter(
...
     }
 
+    @Binds
+    JsPluginRegistration bindJsPlugins(JsPlugins plugins);
+
     @Provides
-    @Named(PluginsModule.JS_PLUGIN_CONSUMER_NAME)
-    static Consumer<JsPlugin> providesJsPluginConsumer(JettyBackedGrpcServer server) {
-        return server.jsPluginConsumer();
+    static JsPlugins providesJsPluginRegistration() {
```
Comment (devinrsmith) [inline server/jetty/src/main/java/io/deephaven/server/jetty/JettyServerModule.java:74]: These also add `JsPlugins` to the public API.

File: server/jetty/src/main/java/io/deephaven/server/jetty/JettyBackedGrpcServer.java (hunk 1 above)
Comment (niloc132) [reply server/jetty/src/main/java/io/deephaven/server/jetty/JettyBackedGrpcServer.java:92]: I bet https://errorprone.info/ can make this fail (or at least has an option for it).

Agreed, I can change this (as discussed briefly in slack) in either direction. I'd lean towards both being public, do you have a preferece?

File: server/jetty/src/main/java/io/deephaven/server/jetty/JettyServerModule.java (hunk 2 above)
Comment (niloc132) [reply server/jetty/src/main/java/io/deephaven/server/jetty/JettyServerModule.java:74]: Ok, seems like making it public is the "better" option?

**Review by devinrsmith — APPROVED** [review]

**Pull request merged**